### PR TITLE
treewide: Cleanup and add compliance workflow

### PR DESCRIPTION
**Pull request by simensrostad**
Cleanup:
- Added compliance workflow.
- Added checkpatch.conf and clang-format files.
- Removed unused files and configurations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,22 +30,18 @@ west update
...
-west build -b thingy91_nrf9160_ns asset_tracker_v2
+west build -b thingy91x/nrf9151/ns app
 ```
 
 If you are using the Powerfoyle solar shield, you can enable power monitoring
 capabilities by using this build command:
 
 ```shell
-west build -b thingy91_nrf9160_ns asset_tracker_v2 -- -DSHIELD=powerfoyle
+west build -b thingy91x/nrf9151/ns app -- -DSHIELD=powerfoyle
```
Comment (jhn-nordic) [inline README.md:40]: i do not think this shield exists for thingy91 X?

Comment (maxd-nordic) [reply README.md:40]: true, this shield only exists for the old thingy

Comment (balaji-nordic) [reply README.md:40]: We planed to remove this in a separate PR. There is a boards folder that should be removed too as we do not intend to support that shield right?

Comment (simensrostad) [reply README.md]: Removed

**Comment by balaji-nordic** [comment]
The build error is due to a zephyr issue. There is a PR that aims to fix this upstream

https://github.com/zephyrproject-rtos/zephyr/pull/72462

**Pull request merged**